File: src/api/MaterialAdvisor/MaterialAdvisor.Application/Models/KnowledgeChecks/Topics/KnowledgeCheckTopic.cs
namespace MaterialAdvisor.Application.Models.Topics;

public class KnowledgeCheckTopic
{
    public Guid Id { get; set; }

    public IEnumerable<KnowledgeCheckQuestion> Questions { get; set; } = [];

    public IEnumerable<LanguageText> Name { get; set; } = [];
}

File: src/app/material-advisor/src/app/models/knowledge-check/topic/KnowledgeCheckTopic.ts
import { LanguageText } from "@shared/models/LanguageText";
import { GUID } from "@shared/types/GUID";
import { KnowledgeCheckQuestion } from "./KnowledgeCheckQuestion";

export interface KnowledgeCheckTopic {
  id: GUID;
  name: LanguageText[];
  questions: KnowledgeCheckQuestion[];
}